dm/docs/RFCS/20211130_enhanced_pre_checker.md

```diff
@@ -0,0 +1,86 @@
+# Enhanced pre-check Design
+
+## Background
+
+Before the DM’s task starts, we should check some items to avoid start-task failures. You can see the details in our [document](https://docs.pingcap.com/tidb-data-migration/stable/precheck#disable-checking-items). In order to allow some users to use DM normally under certain circumstances, we can also set ignore-check-items in task config to ignore some items that you don’t want to check. Now, we find some shortcomings in regard to check-items.
+
+### Bad user habits
+
+We allow users to ignore all check-items, in which case the user's authority is too large to perform unexpected operations. 
+
+### Too much time overhead
+
+If we have a large number of tables in source, we will take too much time in checking table schema, sharding table consistency and sharding table auto increment key.
+
+### Inadequate check
+
+* Now we check it by `mapping` which is deprecated. If we don't set the `mapping` and don't ignore the **auto_increment_ID**, the pre-check will report an error. PS: **auto_increment_ID** is only checked when **schema_of_shard_tables** is true.
+* Dump privilege only checks RELOAD and SELECT. However, Dumpling supports different [consistency configurations](https://docs.pingcap.com/tidb/stable/dumpling-overview#adjust-dumplings-data-consistency-options), which need more privileges.
+* If online-ddl is set by true and a DDL is in online-ddl stage, DM will have a problem in all mode. Specifically, ghost table has been created, is executing the DDL, but is not renamed yet. In this case, DM will report an error when the ghost table is renamed after the dump phase. You can learn more about online-ddl [here](https://docs.pingcap.com/tidb-data-migration/stable/feature-online-ddl).
+* For schema_of_shard_tables, whatever pessimistic task and optimistic task, we all check it by comparing all sharding tables’ structures for consistency simply. For optimistic mode, we can do better.
+
+## Proposal
+
+### Speed ​​up check
+
+1. Support concurrent check
+    - table_schema
+    - schema_of_shard_tables
+2. Use mydumper.threads as **source_connection_concurrency**, which should update in our document.
+
+#### How to speed up?
+
+Since every checker is concurrent, we can split tables to **source_connection_concurrency** part, and create a checker for every part. 
```

Comment:
that's too much details, I prefer we remove this section because any concurrent dispatching implementation is OK. For example, many consumer reading a channel when it is idle.